filecoindot_proofs/src/verify.rs

```diff
@@ -0,0 +1,57 @@
+use crate::errors::Error;
+use cid::Cid;
+
+/// Verify id provided by the against the proof and the root
+/// For the proof, it should be generated by the generate_proof method
+/// in the HAMT mod. In this case, the root should match the
+/// first element in the proof.
+pub fn verify(root: &Cid, proof: Vec<Vec<u8>>, cid: &Cid) -> Result<(), Error> {
+    if proof.is_empty() || proof[0] != root.to_bytes() {
+        return Err(Error::VerificationFailed);
+    }
+
+    let bytes = cid.to_bytes();
+    match proof.iter().find(|p| **p == bytes) {
+        Some(_) => Ok(()),
+        None => Err(Error::VerificationFailed),
+    }
```

Comment:
this only if the proof node is included in the list,
we need to ensure that the proof is actually correct, by traversing the whole proof nodes again and check that all hashes match as expected

Reply:
Ok ok, I'll add a struct that handles this